### ISSUE TITLE
feature: disable auto-registration for provider logins when registrations are disabled

### ISSUE THREAD

**Issue by ivanskodje** [uxbox/uxbox#4283]
### Is your feature request related to a problem? Please describe.

The problem is that I have no way to disable registrations while still allowing provider logins. 
If I disable providers, everything works perfectly, but I would like to allow use of providers for invitations only, to make it easier for them to use penpot, and it makes them feel safer.

### Describe the solution you'd like.

The solution is to simply NOT add and register unknown users logged in with providers when we have a disabled registrations flag. 
 
Perhaps creating a `disable-provider-registrations` flag would be appropriate for this, since `disable-registration` is used for the internal login/registration

### Additional context

Currently the _hack_ to prevent new registrations via providers is to add a domain whitelist for registrations, but doing so will still prevent anyone from trying to register when they get an invite. 

In other words, we cannot invite anyone and make them "register" with the invitation.

Frontend Flags:
`- PENPOT_FLAGS=disable-registration enable-login-with-password enable-login-with-google enable-email-verification enable-smtp`

Backend Flags:
`- PENPOT_FLAGS=disable-registration enable-login-with-password enable-smtp enable-prepl-server enable-login-with-google enable-email-verification`

`- PENPOT_REGISTRATION_DOMAIN_WHITELIST="doesnotexist.example.com"`

**Comment by niwinz** [comment]
Hello @ivanskodje 

I think what you say makes sense, I would even say that **disable-registration** should by default disable registration via oidc providers as well, just like it disables normal registration. What do you think about this? This way we don't have to add a new flag and we make the flag we already have work for all cases. From my point of view it makes sense.

On the other hand, I would like to take advantage of this to tell you that with 2.0 some important changes related to OIDC will be included. With those changes:

- registration through OIDC, in case the provider cannot tell if the email is verified (clear example: github), we will proceed to go through email validation.
- if you have registered with Google for example but then try to log in with GitHub (even if you have the same email on both sites), it will not allow you; It is a safety measure, because not all providers properly report

**Comment by june128** [comment]
@niwinz Having the `disable-registration` flag also simply disable OIDC registrations breaks the use-case, where one wants to have a private penpot instance, where OIDC registrations from e.g. a private Keycloak should work, but one doesn't want local accounts to get created. With the flag disabling both, you can either break sign-ups for everyone - also the Keycloak - or have the possibility of random users registering an account.
I believe #4882 (having this use-case now broken) was created in response to this issue getting resolved by 81b52d7170245aec974f1a17c1bab936150819e8 (pls correct me, if I'm wrong).
